homeassistant/auth/__init__.py

```diff
@@ -46,6 +57,66 @@ def generate_secret(entropy: int = 32) -> str:
     return binascii.hexlify(os.urandom(entropy)).decode('ascii')
 
 
+class InvalidAuth(HomeAssistantError):
+    """Raised when we encounter invalid authentication."""
+
+
+class AuthModule:
```

Comment:
We should not call it Module. It's way too generic. Maybe TwoFactorModule.

Also, this class should be moved into that folder `auth/two_factor_module/__init__.py`

Reply:
I have used MFA, so I will change AuthModule to MultiFactorAuthModule and change folder to auth/mfa_module